addon/services/cookies.js

```diff
@@ -73,35 +73,25 @@ export default Ember.Service.extend({
   },
 
   _writeDocumentCookie(name, value, options = {}) {
-    let cookie = `${name}=${value}`;
-
-    if (!isEmpty(options.domain)) {
-      cookie = `${cookie}; domain=${options.domain}`;
-    }
-    if (typeOf(options.expires) === 'date') {
-      cookie = `${cookie}; expires=${options.expires.toUTCString()}`;
-    }
-    if (!isEmpty(options.maxAge)) {
-      cookie = `${cookie}; max-age=${options.maxAge}`;
-    }
-    if (!!options.secure) {
-      cookie = `${cookie}; secure`;
-    }
-    if (!isEmpty(options.path)) {
-      cookie = `${cookie}; path=${options.path}`;
-    }
-    this.set('_document.cookie', cookie);
+    let serializedCookie = this._serializeCookie(name, value, options);
+    this.set('_document.cookie', serializedCookie);
   },
 
   _writeFastBootCookie(name, value, options = {}) {
+    let responseHeaders  = this.get('_fastBoot.response.headers');
+    let serializedCookie = this._serializeCookie(...arguments);
+
     if (!isEmpty(options.maxAge)) {
       options.maxAge = options.maxAge * 1000;
     }
 
     this._cacheFastBootCookie(...arguments);
 
-    let response = this.get('_fastBoot._fastBootInfo.response');
-    response.cookie(name, value, options);
+    if (responseHeaders.has('cookie')) {
+      responseHeaders.append('cookie', serializedCookie);
+    } else {
+      responseHeaders.set('cookie', serializedCookie);
+    }
```

Comment:
Good point. I think functionality for writing cookies belongs in the `FastBootResponse` then though - instead of having to manually set cookie headers. I guess that's the goal of `FastBootResponse` - to abstract little differences between express and other server side frameworks?

/cc @danmcclain